src/lib/pickles/wrap_verifier.ml

```diff
@@ -191,15 +193,32 @@ struct
       -> (Inner_curve.t index', n) Vector.t
       -> Inner_curve.t index' =
     let open Tuple_lib in
-    let map = Plonk_verification_key_evals.map in
+    let map = Plonk_verification_key_evals.in_circuit_map in
     let map2 = Plonk_verification_key_evals.map2 in
     fun bs keys ->
       let open Field in
       Vector.map2
         (bs :> (Boolean.var, n) Vector.t)
         keys
         ~f:(fun b key -> map key ~f:(fun g -> Double.map g ~f:(( * ) (b :> t))))
-      |> Vector.reduce_exn ~f:(map2 ~f:(Double.map2 ~f:( + )))
+      |> Vector.reduce_exn
+           ~f:
+             (let f = Double.map2 ~f:( + ) in
+              let _add_opt x y =
+                match (x, y) with
+                | Opt.Some x, Opt.Some y ->
+                    Opt.some (x + y)
+                | o, Opt.None | Opt.None, o ->
+                    o
+                | Opt.Maybe (_b, x1), Opt.Maybe (_b2, x2) ->
+                    Opt.maybe _b2 (x1 + x2)
+                | Opt.Some x, Opt.Maybe (_b, y) | Opt.Maybe (_b, y), Opt.Some x
+                  ->
+                    Opt.some (x + y)
+                (* FIXME:  Look at the definitions of these guys. Not sure what
+                   to do here since the shape of b is not even known *)
+              in
+              map2 ~f ~f_opt:(fun _ _ -> assert false) )
```

Comment:
Won't this trigger when we have the optional commitments turned on a vk?

Reply:
Indeed. Thanks for having caught that 🙇‍♂️